### PR TITLE
Extract missing modules translations

### LINKED ISSUE #28418: Allow to translate new wordings in core modules
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

I forked a core modules. I added a feature that required a ne wording to translate.

When I tried to translate it in the BO `International - Translations - Installed Modules translations`, the new wordings did not appear.

This behavior is specific to core modules.

### Describe the solution you'd like

I'd like to be able to translate new wordings in a fork of a core module using the BO.

### Alternatives you've considered

I found a workaround: 

1. Delete (and backup) the files pertaining to the module in the folder `/app/Resources/translations/default`.
2. Translate the new wordings in the BO and export the translation files
3. Restore the files deleted above.

### Additional context

See https://github.com/PrestaShop/ps_emailalerts/pull/113#discussion_r864592776.
Discussed here: https://prestashop.slack.com/archives/CTKHSPWPM/p1651654441661609

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | When displaying the modules translations we combine the XLF default ones with the extracted ones from templates We do this even for native modules so that wordings present in templates but absent from the XLK catalog are still visible. Also this required updating the dependency from the TranslationToolsBundle because it had a bug in the extractor see this PR https://github.com/PrestaShop/TranslationToolsBundle/pull/107
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #28418
| Related PRs       | Fix in bundle https://github.com/PrestaShop/TranslationToolsBundle/pull/107
| How to test?      | See below for more details
| Possible impacts? | Check that module's translation interface still works well and does not miss any wordings.

#### Step 1

First test with the 8.0.x branch and install [this module](https://github.com/PrestaShop/PrestaShop/files/9463723/ps_linklist.zip) on your shop, it is a modification from the original `ps_linklist` native module that includes three new wordings:
- `New super link block` (domain `Modules.Linklist.Admin`)
- `My link grid` (domain `Modules.Linklist.Admin`)
- `My link block` (domain `Modules.Linklist.Shop`)

Then try to edit the module's translation (see the following video to do so):

https://user-images.githubusercontent.com/13801017/187744581-f2a3d370-1115-49eb-99e7-c397f995be64.mp4

With the 8.0.x branch, you should see none of those new wordings:

Capture d’écran 2022-08-31 à 19 47 23

#### Step 2

Now switch to this PR's branch but don't run `composer install` yet (you may need to remove Symfony cache though)
Try to edit the wordings again, you should now see one new wording for Shop and one new wording for Admin. `My link grid` is missing:

Capture d’écran 2022-08-31 à 19 45 48

This validates the fix in the Core code from this PR, new wordings are now extracted all the time even if they are not in the XLF catalogue. But one wording is still missing, this is fixed in the bundle, now proceed to step 3.

#### Step 3

Run `composer install` to get the update from the translation bundle and edit the wordings again, you should now see correctly the three new wordings. This validates the fix in the bundle from this PR https://github.com/PrestaShop/TranslationToolsBundle/pull/107

Capture d’écran 2022-08-31 à 19 25 01

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -195,6 +195,12 @@
             "vendor/symfony/symfony/src/Symfony/Component/VarDumper/Resources/functions/dump.php"
         ]
     },
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "git@github.com:jolelievre/TranslationToolsBundle.git"
```
Comment (matks) [inline composer.json]: This is not something we can merge 😄

Comment (jolelievre) [reply composer.json]: Yeah I'm aware, this was for the test The bundle has been merged I'm gonna update this

Comment (jolelievre) [reply composer.json]: @matks it's ok now!

**Review by matks — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by khouloudbelguith — APPROVED** [review]
Hi @jolelievre,

I checked 

Automatic tests => OK

- Translation BO with English language & Italian language => ok
- Translation FO => OK
- Translation Installed module => OK
- Translation Email subject => OK
- Translation Email body => OK
- Translation Other => OK

1. I tried the [ps_linklist](https://github.com/PrestaShop/PrestaShop/files/9463723/ps_linklist.zip) module => OK
2. I tried the ps_mbo module as commented here: https://github.com/PrestaShop/TranslationToolsBundle/issues/108 => OK
3. I tried some manual modification on the ps_imageslider module => OK [ps_imageslider.zip](https://github.com/PrestaShop/PrestaShop/files/9528377/ps_imageslider.zip)

I attached a screen record

https://user-images.githubusercontent.com/16067358/189166644-0780c8ff-72ee-4829-980a-f2b8fc9a245a.mp4

Thanks!

**Comment by atomiix** [comment]
Thanks @jolelievre & @khouloudbelguith

**Pull request merged**